.travis.yml

```diff
@@ -0,0 +1,87 @@
+language: python
+sudo: false
+cache:
+  directories:
+    - $HOME/.ccache  # https://github.com/travis-ci/travis-ci/issues/5853
+  pip: true
+
+addons:
+  apt:
+    update: true
+    packages:
+      - csh
+      - libhdf5-serial-dev
+      - gfortran
+      - libtool
+
+env:
+  global:
+    YT_DATA_DIR=$HOME/yt_test
+
+before_install:
+  - |
+    echo "backend : Agg" > $HOME/matplotlibrc
+    export MATPLOTLIBRC=$HOME
+    # Download test data
+    mkdir -p $YT_DATA_DIR
+    pushd $YT_DATA_DIR
+    wget --quiet http://yt-project.org/data/IsolatedGalaxy.tar.gz
+    tar xzf IsolatedGalaxy.tar.gz
+    popd
+
+install:
+  - |
+    # setup environment
+    ccache -s
+    # Upgrade pip and setuptools and wheel to get clean install
+    pip install --upgrade pip
+    pip install --upgrade wheel
+    pip install --upgrade setuptools
+    # Install dependencies
+    pip install numpy cython h5py matplotlib fastcache flake8 pytest yt
+    # Install Grackle
+    ./configure
+    cd src/clib
+    make machine-linux-gnu
+    if [[ $TRAVIS_BUILD_STAGE_NAME = "Omp" ]]; then
+      make omp-on
+    fi
+    make
+    mkdir -p $HOME/local
+    make install
+    export LD_LIBRARY_PATH=$HOME/local/lib:$LD_LIBRARY_PATH
+    if [[ $TRAVIS_BUILD_STAGE_NAME != "Omp" ]]; then
+      cd ../python
+      python setup.py develop
+    fi
+    cd ../
+
+jobs:
+  include:
+    - stage: tests
+      name: "tests with Python 2.7"
+      python: 2.7
+      script: py.test python/tests
+
+    - stage: tests
+      name: "tests with Python 3.5"
+      python: 3.5
+      script: py.test python/tests
+
+    - stage: tests
+      name: "tests with Python 3.6"
+      python: 3.6
+      script: py.test python/tests
```

Comment:
From my experience with unyt I think it should be straighforward to migrate this to use pyenv, which will separate you from needing to depend on travis' python setup, which is slow to keep up with upstream so you can also test python3.7 here.

Reply:
Maybe this will help with the test failure issues.